app/src/main/java/com/asfoundation/wallet/ui/gamification/LevelsAdapter.kt

```diff
@@ -59,15 +65,37 @@ class LevelsAdapter(private val context: Context,
     }
   }
 
+  /**
+   * set the content to display regarding the several levels
+   * with this one can update the view holders several times
+   */
+  fun setLevelsContent(_hiddenLevels: List<LevelModel>, _shownLevels: List<LevelModel>,
+                       _amountSpent: BigDecimal, _currentLevel: Int,
+                       _nextLevelAmount: BigDecimal?) {
+    activeLevelList.clear()
+    hiddenLevels = _hiddenLevels
+    shownLevels = _shownLevels
+    amountSpent = _amountSpent
+    currentLevel = _currentLevel
+    nextLevelAmount = _nextLevelAmount
+    if (reachedLevelsToggled) {
+      activeLevelList.addAll(hiddenLevels)
+      activeLevelList.addAll(shownLevels)
+    } else activeLevelList.addAll(shownLevels)
+    notifyDataSetChanged()
+  }
+
   fun toggleReachedLevels(show: Boolean) {
     if (show) {
+      reachedLevelsToggled = true
       if (currentLevel != 0) {
```

Comment:
Not yours, but at the time it seemed like a good solution at the time. But the better solution would be to get the positon of the CurrentLevelItem, if it's different than 0 then... if(getCurrentLevelItemPosition != 0)